### PR TITLE
GuestConfig PS Cmdlets: Fix Inconsistencies in compliance status

### PR DESCRIPTION
**Pull request by aashish1987**
## Description

1. Fix inconsistency of compliance status due to update paramters.
2. Fix retrieving extra policy data with same metadata properties.

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/GuestConfiguration/GuestConfiguration.Test/ScenarioTests/GetVmGuestPolicyStatusHistoryTests.ps1 [hunk 1]
```diff
@@ -32,8 +32,8 @@ Get guest configuration policy status history by Vm name scope
 #>
 function Get-AzVMGuestPolicyStatusHistory-VmNameScope_Custom
 {
-	$rgName = "amits-test"
-	$vmName = "amits-winser6"
+	$rgName = "aashishCustomrole7ux"
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration.Test/ScenarioTests/GetVmGuestPolicyStatusHistoryTests.ps1]: You should not hard code resource group and vm names - you should use the existing infrastructure to generate a random name for this test

File: src/GuestConfiguration/GuestConfiguration.Test/ScenarioTests/GetVmGuestPolicyStatusTests.ps1 [hunk 2]
```diff
@@ -32,8 +32,8 @@ Get guest configuration policy report by Vm name scope for custom policy
 #>
 function Get-AzVMGuestPolicyStatus-VmNameScope_Custom
 {
-	$rgName = "amits-test"
-	$vmName = "amits-winser6"
+	$rgName = "aashishCustomrole7ux"
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration.Test/ScenarioTests/GetVmGuestPolicyStatusTests.ps1]: same comment - users should not have to update these names whenever they re-record these tests

File: src/GuestConfiguration/GuestConfiguration.Test/GuestConfiguration.Test.csproj [hunk 3]
```diff
@@ -12,6 +12,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Management.GuestConfiguration" Version="1.1.0" />
+    <PackageReference Include="Microsoft.Azure.Management.PolicyInsights" Version="2.0.2" />
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration.Test/GuestConfiguration.Test.csproj]: This is going to cause problems when the PolicyInsights client is updated by that team.  One option would be to move these cmdlets into the PolicyInsights module.  Another would be to capture a common library for policy insights here: https://github.com/Azure/azure-powershell-common/tree/master/src

File: src/GuestConfiguration/GuestConfiguration/Common/GuestConfigurationCmdletBase.cs [hunk 4]
```diff
@@ -49,6 +51,11 @@ public abstract class GuestConfigurationCmdletBase : AzureRMCmdlet
         /// </summary>
         private IPolicyClient _policyClient;
 
+        /// <summary>
+        /// Policy insights client
+        /// </summary>
+        private IPolicyInsightsClient _policyInsightsClient;
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/Common/GuestConfigurationCmdletBase.cs]: You may not create a cross-dependency between yoiur cmdlet and the policy insights management library - the same solutions outlined above would work (haveing a captured, common version of policy insights, or moving these cmdlets into the PolicyInsights module.

File: src/GuestConfiguration/GuestConfiguration/GuestConfiguration.csproj [hunk 5]
```diff
@@ -12,6 +12,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Management.GuestConfiguration" Version="1.1.0" />
+    <PackageReference Include="Microsoft.Azure.Management.PolicyInsights" Version="2.0.2" />
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/GuestConfiguration.csproj]: See above.  Only the PolicyInsights module may take a dependency on the PolicyInsights management client

File: src/GuestConfiguration/GuestConfiguration/Models/PolicyData.cs [hunk 6]
```diff
@@ -41,6 +42,8 @@ public PolicyData()
 
         }
 
+        public string PolicySetDefinitionName { get; set; }
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/Models/PolicyData.cs]: Also, please add a changelog entriy describing these changes - this is the trigger that determines when a new module will be released.

**Comment by markcowl** [comment]
@aashish1987 Please update this PR as specified.  Code complete for the current release is today

File: src/GuestConfiguration/GuestConfiguration.Test/ScenarioTests/GetVmGuestPolicyStatusHistoryTests.ps1 (hunk 1 above)
Comment (aashish1987) [reply src/GuestConfiguration/GuestConfiguration.Test/ScenarioTests/GetVmGuestPolicyStatusHistoryTests.ps1]: I will create a new PR to change the way tests are written for GCRP cmd-lets.
Wont fix in this PR.

File: src/GuestConfiguration/GuestConfiguration.Test/ScenarioTests/GetVmGuestPolicyStatusTests.ps1 (hunk 2 above)
Comment (aashish1987) [reply src/GuestConfiguration/GuestConfiguration.Test/ScenarioTests/GetVmGuestPolicyStatusTests.ps1]: I will create a new PR to update how tests have been written for GCRP PowerShell cmdets.
Wont fix in this PR.

File: src/GuestConfiguration/GuestConfiguration/GuestConfiguration.csproj (hunk 5 above)
Comment (aashish1987) [reply src/GuestConfiguration/GuestConfiguration/GuestConfiguration.csproj]: Updated.

File: src/GuestConfiguration/GuestConfiguration/Models/PolicyData.cs (hunk 6 above)
Comment (aashish1987) [reply src/GuestConfiguration/GuestConfiguration/Models/PolicyData.cs]: Updated

**Comment by cormacpayne** [comment]
Waiting for this PR to be merged before continuing: https://github.com/Azure/azure-powershell/pull/9460

**Comment by cormacpayne** [comment]
@aashish1987 ping on the comments above

**Pull request merged**